v2/app/src/main/java/com/google/firebase/example/makeitso/ui/settings/SettingsScreen.kt

```diff
@@ -0,0 +1,138 @@
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.rememberTopAppBarState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.google.firebase.example.makeitso.R
+import com.google.firebase.example.makeitso.ui.shared.CenterTopAppBar
+import com.google.firebase.example.makeitso.ui.shared.StandardButton
+import com.google.firebase.example.makeitso.ui.theme.DarkBlue
+import com.google.firebase.example.makeitso.ui.theme.DarkGrey
+import com.google.firebase.example.makeitso.ui.theme.LightRed
+import kotlinx.serialization.Serializable
+
+@Serializable
+object SettingsRoute
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+fun SettingsScreen(
+    openHomeScreen: () -> Unit,
+    openSignInScreen: () -> Unit,
+    viewModel: SettingsViewModel = hiltViewModel()
+) {
+    val isAnonymous by viewModel.isAnonymous.collectAsStateWithLifecycle()
+    val scrollBehavior = TopAppBarDefaults.pinnedScrollBehavior(rememberTopAppBarState())
+
+    LaunchedEffect(true) {
+        viewModel.loadCurrentUser()
+    }
+
+    Scaffold(
+        topBar = {
+            CenterTopAppBar(
+                title = stringResource(R.string.settings),
+                scrollBehavior = scrollBehavior
+            )
+        },
+    ) { innerPadding ->
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(
+                    top = innerPadding.calculateTopPadding(),
+                    start = 4.dp,
+                    end = 4.dp,
+                    bottom = 4.dp
+                )
+        ) {
+            Spacer(Modifier.size(24.dp))
+
+            if (isAnonymous) {
+                StandardButton(R.string.sign_in) {
+                    openSignInScreen()
+                }
+            } else {
+                StandardButton(R.string.sign_out) {
+                    viewModel.signOut(openHomeScreen)
+                }
+
+                Spacer(Modifier.size(16.dp))
+
+                DeleteAccountButton(openHomeScreen, viewModel)
+            }
+        }
+    }
+}
+
+@Composable
+fun DeleteAccountButton(
+    openHomeScreen: () -> Unit,
+    viewModel: SettingsViewModel = hiltViewModel()
+) {
+    val showDeleteAccountDialog = remember { mutableStateOf(false) }
+
+    StandardButton(R.string.delete_account) {
+        showDeleteAccountDialog.value = true
+    }
+
+    if (showDeleteAccountDialog.value) {
```

Comment:
nit: you can initialize the `showDeleteAccountDialog` with:

```kotlin
var showDeleteAccountDialog by remember { mutableStateOf(false) }
```

So that this if statement becomes:
```kotlin
if (showDeleteAccountDialog) {
```

Reply:
Great catch - done!